sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UsingCookies_NetCore.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        void Responses(HttpResponse response)
+        {
+            var value = "";
+
+            // Response headers
+            response.Headers.Add("Set-Cookie", ""); // Noncompliant
+            response.Headers["Set-Cookie"] = ""; // Noncompliant
+            value = response.Headers["Set-Cookie"]; // Noncompliant
+
+            // Not the Set-Cookie header
+            response.Headers.Add("something", "");
+            response.Headers["something"] = value;
+            value = response.Headers["something"];
+
+            // Response headers as variable
+            var responseHeaders = response.Headers;
+            responseHeaders.Add("Set-Cookie", ""); // Noncompliant
+            responseHeaders["Set-Cookie"] = ""; // Noncompliant
+            value = responseHeaders["Set-Cookie"]; // Noncompliant
+
+            // Response cookies as property
+            response.Cookies.Append("", ""); // Noncompliant
+            response.Cookies.Append("", "", new CookieOptions { }); // Noncompliant
+
+            // Response cookies as variable
+            var responseCookies = response.Cookies;
+            responseCookies.Append("", ""); // Noncompliant
+            responseCookies.Append("", "", new CookieOptions { }); // Noncompliant
```

Comment:
Maybe include a call to a method that shouldn't be checked e.g. _Delete_ is specifically mentioned in the RSPEC.